client-js/app/scripts/registry.ts

```diff
@@ -9,182 +11,10 @@ export interface RegistryListing {
   vals: Array<string>;
 }
 
-export interface RegistryApi {
```

Comment:
the methods just have to implemented in the class, but yes, it does mean repeating the type signature of the method, which here looks like a lot because the method bodies themselves are so short.